mypy/checkexpr.py

```diff
@@ -1610,10 +1609,13 @@ def analyze_super(self, e: SuperExpr, is_lvalue: bool) -> Type:
                         return AnyType()
                     if not self.chk.in_checked_function():
                         return AnyType()
-                    return analyze_member_access(e.name, self_type(e.info), e,
-                                                 is_lvalue, True, False,
-                                                 self.named_type, self.not_ready_callback,
-                                                 self.msg, base, chk=self.chk)
+                    selftype = self.chk.binder.get(NameExpr('__SelfType'))
+                    return analyze_member_access(name=e.name, typ=fill_typevars(e.info), node=e,
+                                                 is_lvalue=False, is_super=True, is_operator=False,
+                                                 builtin_type=self.named_type,
+                                                 not_ready_callback=self.not_ready_callback,
+                                                 msg=self.msg, override_info=base, chk=self.chk,
+                                                 report_type=selftype)
```

Comment:
I think will call it "actual_self" consistently throughout the code, replacing both report_type and actual_self. What do you think? There seem to be no standard or self-explanatory name for this thing. "the static type of the first argument" - except the fact that in mypy `Argument` is actually a parameter, so we self_arg_type is just as confusing.